core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java

```diff
@@ -68,12 +69,12 @@ public class OAuth2TokenExchangeResource
     private final ScheduledExecutorService timeoutExecutor;
 
     @Inject
-    public OAuth2TokenExchangeResource(OAuth2TokenExchange tokenExchange, OAuth2Service service, DispatchExecutor executor)
+    public OAuth2TokenExchangeResource(OAuth2TokenExchange tokenExchange, OAuth2Service service, @ForOAuth2 ExecutorService executor, @ForOAuth2 ScheduledExecutorService scheduledExecutor)
     {
         this.tokenExchange = requireNonNull(tokenExchange, "tokenExchange is null");
         this.service = requireNonNull(service, "service is null");
-        this.responseExecutor = executor.getExecutor();
-        this.timeoutExecutor = executor.getScheduledExecutor();
+        this.responseExecutor = listeningDecorator(requireNonNull(executor, "executor is null"));
+        this.timeoutExecutor = requireNonNull(scheduledExecutor, "scheduledExecutor is null");
```

Comment:
nit: Just curious would it make more sense to have a bit more explicit assertions, f.e.
```suggestion
        this.responseExecutor = listeningDecorator(requireNonNull(executor, "responseExecutor is null"));
        this.timeoutExecutor = requireNonNull(scheduledExecutor, "timeoutExecutor is null");
```